src/pipeline.rs

```diff
@@ -154,9 +217,11 @@ fn parse_json_line(
 }
 
 /// Parse a line of TSV and read its columns into a vector for processing.
-fn parse_tsv_line(line: &str) -> Vec<SmallVec<[&str; SMALL_VECTOR_SIZE]>> {
+fn parse_tsv_line(line: &str) -> Vec<SmallVec<[String; SMALL_VECTOR_SIZE]>> {
```

Comment:
Random idea to try now or later: This increases amount of allocation, have you tried using smartstring::alias::String? Maybe this would decrease amount of allocations (but it's probably not an issue here anyway).

Reply:
Didnt check it. Imho lets not prematurely optimize